packages/frontend/core/src/components/affine/auth/oauth.tsx

```diff
@@ -46,19 +45,25 @@ export function OAuth() {
 }
 
 function OAuthProvider({ provider }: { provider: OAuthProviderType }) {
+  const auth = useService(AuthService);
   const { icon } = OAuthProviderMap[provider];
 
   const onClick = useCallback(() => {
-    let oauthUrl =
-      (BUILD_CONFIG.isElectron ? BUILD_CONFIG.serverUrlPrefix : '') +
-      `/oauth/login?provider=${provider}`;
-
-    if (BUILD_CONFIG.isElectron) {
-      oauthUrl += `&client=${appInfo?.schema}`;
+    async function preflight() {
```

Comment:
the logic can't be async otherwise safari will block popup

Reply:
fixed